examples/05_tool_inference_lock/ex_tool_inference_lock/inference_lock_helper.rb

```diff
@@ -0,0 +1,122 @@
+module Examples
+  module ToolInferenceLock
+    # Helper object to lock tool inference using Shift or arrow keys.
+    class InferenceLockHelper
+      # Create InferenceLockHelper object.
+      #
+      # @para tool [Object] The SketchUp Ruby tool using this helper object.
+      # @param active_ip [Sketchup::InputPoint]
+      # @param reference_ip [Sketchup::InputPoint]
+      def initialize(tool, active_ip, reference_ip = active_ip)
+        @tool = tool
+        @active_ip = active_ip
+        @reference_ip = reference_ip
+
+        @axis_lock = nil
+        @mouse_x = nil
+        @mouse_y = nil
+      end
+
+      # Call this method from +onKeyDown+.
+      def handle_keydown(key, view)
+        if key == CONSTRAIN_MODIFIER_KEY
+          try_lock_constraint(view)
+        else
+          try_lock_axis(key, view)
+        end
+
+        # Emulate mouse move to update InputPoint picking.
+        @tool.onMouseMove(0, @mouse_x, @mouse_y, view) if @mouse_x && @mouse_y
+        view.invalidate
+      end
+
+      # Call this method from +onKeyUp+.
+      def handle_keyup(key, view)
```

Comment:
Maybe `on_keyup`? Matching naming patterns elsewhere in the API.

Reply:
Done